visual-js/visual/src/utils.ts

```diff
@@ -103,24 +112,43 @@ export const getDiffingOptions = <T>(
  * @param resolveItem A callback to resolve an element and gather the required data for the API.
  */
 export const parseRegionsForAPI = async <T>(
-  ignore: (T | RegionIn | RegionType<T> | Promise<RegionIn>)[],
+  ignore: (
+    | T
+    | RegionIn
+    | RegionType<T>
+    | Promise<RegionIn>
+    | IgnoreSelectorIn
+  )[],
   resolveItem: (
     item: T | Promise<RegionIn>,
-  ) => Promise<(RegionIn | ElementIn)[]>,
+  ) => Promise<(RegionIn | ElementIn | IgnoreSelectorIn)[]>,
 ): Promise<{
   ignoreRegions: RegionIn[];
   ignoreElements: ElementIn[];
+  ignoreSelectors: IgnoreSelectorIn[];
 }> => {
-  const promisedIgnorables: Promise<(RegionIn | ElementIn)[]>[] = ignore.map(
-    async (itemOrRegion): Promise<Array<RegionIn | ElementIn>> => {
-      const { item, diffingOptions } = isRegionType(itemOrRegion)
+  const promisedIgnorables: Promise<
+    (RegionIn | ElementIn | IgnoreSelectorIn)[]
+  >[] = ignore.map(
+    async (
+      itemOrRegionOrSelector,
+    ): Promise<Array<RegionIn | ElementIn | IgnoreSelectorIn>> => {
+      const { item, diffingOptions } = isRegionType(itemOrRegionOrSelector)
```

Comment:
Might be cleaner with if / else instead of nested ternary statements?

Reply:
added if else statement